.github/workflows/pull_request.yml

```diff
@@ -31,7 +31,7 @@ jobs:
 
       - name: Build
         run: |
-          yarn --frozen-lockfile --prefer-offline
```

Comment:
So, removing the frozen-lockfile allows yarn to update dependencies?  Is this desirable?

Reply:
My (current) understanding is that this is a function of the build area caching. We never used to have the --frozen-lockfile argument (see https://github.com/datacite/bracco/blob/e2f05955e691226fd914cf03abdba78f9eeaae1d/.github/workflows/deploy.yml), it was something I added when trying to speed up the build process.

The reason for removing it is that I updated the cache action to also restore older versions of the cache if there was no hit for the current lockfile hash (https://github.com/datacite/bracco/blob/91555b690dc04f9c306dfb73f6fc7e039df88f14/.github/workflows/pull_request.yml#L30), meaning that as much of the workspace could be reused as possible.

The practical effect of this is that when the lockfile changes, the Yarn cache that is restored will not match the lockfile exactly, and so throws an error. I don't _think_ that any package dependencies actually get upgraded beyond what is in the lockfile already as a result of this, but perhaps I should do a bit more testing and double check that assumption.

Reply:
Thinking about it, there's a possibility it could also be that I accidentally committed the wrong version of my local lockfile. Either way, I'll check and confirm what's going on before merge.

Reply:
Cool

Reply:
More investigation showed that I was also getting this error locally when trying to use `--frozen-lockfile`, even though a "free" yarn run was showing "No changes" and not modifying the lockfile at all.

Turns out this is a weird interaction in Yarn with the subdependency pinning that Yarn doesn't handle/log well (at least in the version we use): https://github.com/yarnpkg/yarn/issues/4778#issuecomment-522575790

I made the change in that comment and it looks like the output was `missingPatterns [ 'acorn@^7.1.1' ]` so I will poke about and see if I can fix that, and then restore the --frozen-lockfile argument.

Reply:
Which version of yarn do we use?

Reply:
Locally I am using 1.22.17 and it looks like the build jobs are running the same. A lot of the dependency resolution stuff was reworked in Yarn2, but that's _definitely_ a switch for a future tech debt week :grin: